infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -844,73 +696,67 @@ public void updateTopology(LogReplicationClusterInfo.TopologyConfigurationMsg to
      * Sink Cluster - Read the shared metadata table and find if data is consistent(set to false if snapshot sync is
      * in the apply phase)
      */
-    @Override
-    public Map<String, LogReplicationMetadata.ReplicationStatusVal> queryReplicationStatus() {
-        Map<String, LogReplicationMetadata.ReplicationStatusVal> clientToReplicationStatusMap = new HashMap<>();
+    public Map<LogReplicationSession, ReplicationStatus> queryReplicationStatus() {
+        Map<LogReplicationSession, ReplicationStatus> replicationStatusMap = new HashMap<>();
 
-        if (localClusterDescriptor == null) {
+        if (topologyDescriptor.getLocalClusterDescriptor() == null) {
             log.warn("Cluster configuration has not been pushed to current LR node.");
-            return clientToReplicationStatusMap;
+            return replicationStatusMap;
         }
 
-        if (localClusterDescriptor.getRole() != ClusterRole.SOURCE &&
-            localClusterDescriptor.getRole() != ClusterRole.SINK) {
-            log.error("Received Replication Status Query in Incorrect Role {}.", localClusterDescriptor.getRole());
-            return clientToReplicationStatusMap;
+        if (topologyDescriptor.getLocalClusterDescriptor().getRole() != ClusterRole.SOURCE &&
+                topologyDescriptor.getLocalClusterDescriptor().getRole() != ClusterRole.SINK) {
+            log.error("Replication Status Query in incorrect role {}", topologyDescriptor.getLocalClusterDescriptor().getRole());
+            return replicationStatusMap;
         }
 
-        // Note: MetadataManager is currently instantiated per remote session.  In a subsequent PR, change to share a
-        // single instance for all remote sessions will be added.  So for now, get all replication statuses using any
-        // 1 metadata manager.
-        if (remoteSessionToMetadataManagerMap.values().iterator().hasNext()) {
-            return remoteSessionToMetadataManagerMap.values().iterator().next().getReplicationStatus();
-        }
-        return clientToReplicationStatusMap;
+        return sessionManager.getReplicationStatus();
     }
 
-    @Override
     public UUID forceSnapshotSync(String clusterId) throws LogReplicationDiscoveryServiceException {
```

Comment:
I think We should support forceSnapshotSync for a session

Reply:
yes, see the TODO in this method.  I think the plan is to modify it completely when ClusterManager changes are available.

Reply:
yes, I had it as a TODO cause I thought not breaking the client side. Still anyways its broken because the queryReplicationStatus interface changed in this PR as well. So, it would be ok changing it here. Currently I was just defaulting to the FULL TABLE session for the given cluster id.

Reply:
done